### PR TITLE
Add a check for metadata from auth plugin

### LINKED ISSUE #8100: Core should check headers added via auth plugin API.
Most of the wrapping languages already have the check but since core has the check for normal headers, the headers from auth plugin should also be checked.

### PR DESCRIPTION
**Pull request by yang-g**
Fixes #8100

### CODE REVIEW

**Comment by jboeuf** [comment]
Looks good. Do you want to add a test for the bin value in a non-bin header? The logic is a little more tricky there.

**Comment by yang-g** [comment]
Added a test for that as well. PTAL Thanks.

File: test/cpp/end2end/end2end_test.cc [hunk 1]
```diff
@@ -1322,6 +1333,40 @@ TEST_P(SecureEnd2endTest, OverridePerCallCredentials) {
...
+TEST_P(SecureEnd2endTest, AuthMetadataPluginValueFailure) {
+  ResetStub();
+  EchoRequest request;
+  EchoResponse response;
+  ClientContext context;
+  context.set_credentials(
+      MetadataCredentialsFromPlugin(std::unique_ptr<MetadataCredentialsPlugin>(
+          new TestMetadataCredentialsPlugin(
+              TestMetadataCredentialsPlugin::kGoodMetadataKey,
+              "With binary \n data", false, true))));
```
Comment (jboeuf) [inline test/cpp/end2end/end2end_test.cc]: I didn't know that `\n` was considered as binary data. Maybe just for headers though.

**Comment by jboeuf** [comment]
Thanks much for the new test. LGTM.

File: test/cpp/end2end/end2end_test.cc (hunk 1 above)
Comment (yang-g) [inline test/cpp/end2end/end2end_test.cc]: Updated the metadata message a bit :)

Comment (jboeuf) [inline test/cpp/end2end/end2end_test.cc]: much clearer. Thanks much!

**Pull request merged**